core/src/test/java/org/elasticsearch/search/suggest/SuggestionTests.java

```diff
@@ -144,22 +144,22 @@ private void doTestFromXContent(boolean addRandomFields) throws IOException {
         }
     }
 
-    /**
-     * test that we throw error if RestSearchAction.TYPED_KEYS_PARAM isn't set while rendering xContent
-     */
-    public void testFromXContentFailsWithoutTypeParam() throws IOException {
-        XContentType xContentType = randomFrom(XContentType.values());
-        BytesReference originalBytes = toXContent(createTestItem(), xContentType, ToXContent.EMPTY_PARAMS, randomBoolean());
-        try (XContentParser parser = createParser(xContentType.xContent(), originalBytes)) {
-            ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser::getTokenLocation);
-            ensureExpectedToken(XContentParser.Token.FIELD_NAME, parser.nextToken(), parser::getTokenLocation);
-            ParsingException e = expectThrows(ParsingException.class, () -> Suggestion.fromXContent(parser));
-            assertEquals(
-                    "Cannot parse object of class [Suggestion] without type information. "
-                    + "Set [typed_keys] parameter on the request to ensure the type information "
-                    + "is added to the response output", e.getMessage());
-        }
-    }
+//    /**
+//     * test that we throw error if RestSearchAction.TYPED_KEYS_PARAM isn't set while rendering xContent
+//     */
+//    public void testFromXContentFailsWithoutTypeParam() throws IOException {
+//        XContentType xContentType = randomFrom(XContentType.values());
+//        BytesReference originalBytes = toXContent(createTestItem(), xContentType, ToXContent.EMPTY_PARAMS, randomBoolean());
+//        try (XContentParser parser = createParser(xContentType.xContent(), originalBytes)) {
+//            ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser::getTokenLocation);
+//            ensureExpectedToken(XContentParser.Token.FIELD_NAME, parser.nextToken(), parser::getTokenLocation);
+//            ParsingException e = expectThrows(ParsingException.class, () -> Suggestion.fromXContent(parser));
+//            assertEquals(
+//                    "Cannot parse object of class [Suggestion] without type information. "
+//                    + "Set [typed_keys] parameter on the request to ensure the type information "
+//                    + "is added to the response output", e.getMessage());
+//        }
+//    }
```

Comment:
I left the test for now, this checkes that we throw an exception when the type information is missing. We won't do that any more if we are always lenient.